agent/src/beerocks/slave/gate/unit_test/gate_test.cpp

```diff
@@ -0,0 +1,121 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "1905_beacon_query_to_vs.h"
+#include <bcl/network/network_utils.h>
+#include <tlvf/CmduMessageTx.h>
+
+#include <gtest/gtest.h>
+
+namespace {
+
+/**
+* const values to be used for both assignment and test
+*/
+
+const auto associated_sta_mac = sMacAddr{0x01, 0x02, 0x03, 0x04, 0x05, 0x06};
+const uint8_t operating_class = 0x73;
+const uint8_t channel_number  = 255;
+const uint8_t channel_1       = 46;
+const uint8_t channel_2       = 31;
+
+/**
+* helper function to fill 1905 values
+*/
+static bool set_1905_values(ieee1905_1::CmduMessageTx &cmdu1905)
+{
+    auto beacon_metrics_query = cmdu1905.getClass<wfa_map::tlvBeaconMetricsQuery>();
+    if (!beacon_metrics_query) {
+        return false;
+    }
+
+    // fill some values
+    beacon_metrics_query->associated_sta_mac() = associated_sta_mac;
+    beacon_metrics_query->operating_class()    = operating_class;
+    // for ap channel report list, this must be set to 255
+    beacon_metrics_query->channel_number() = channel_number;
+
+    auto ap_channel_list = beacon_metrics_query->create_ap_channel_reports_list();
+    if (!ap_channel_list) {
+        return false;
+    }
+
+    ap_channel_list->alloc_ap_channel_report_list(3);
+    // the first is operating class, which is currently ignored system wide
+    *ap_channel_list->ap_channel_report_list(0) = operating_class;
+    *ap_channel_list->ap_channel_report_list(1) = channel_1;
+    *ap_channel_list->ap_channel_report_list(2) = channel_2;
+
+    beacon_metrics_query->add_ap_channel_reports_list(ap_channel_list);
+
+    return true;
+}
+
+TEST(gate_beacon_query_test, loading_vs_specific_from_1905_beacon_query)
+{
+    // preperation for beerocks
+
+    // class list
+
+    constexpr const uint16_t BUF_SIZE_BEEROCKS = 10000; // just an arbitrary size
```

Comment:
No need for `const` after `constexpr`
UPPER_CASE is for #define's

Reply:
done.